src/pages/Armies/EditArmyPage.test.jsx

```diff
@@ -0,0 +1,65 @@
+import React from 'react';
+import Enzyme, { mount } from 'enzyme';
+import EnzymeAdapter from 'enzyme-adapter-react-16';
+import firebase from 'firebase/app';
+import { findByTestAttr } from '../../utilities/testutils';
+import firestore from '../../utilities/mockFirestore';
+import EditArmyPage from './EditArmyPage';
+import store from '../../Redux/store';
+import SquadsChange from '../../Redux/actions/SquadsChange';
+import UnitsChange from '../../Redux/actions/UnitsChange';
+import UserChange from '../../Redux/actions/UserChange';
+import RolesChange from '../../Redux/actions/RolesChange';
+import CodexChange from '../../Redux/actions/CodexChange';
+import EquipmentChange from '../../Redux/actions/EquipmentChange';
+
+firebase.firestore = firestore;
+Enzyme.configure({ adapter: new EnzymeAdapter() });
+
+const mockSetisLoading = jest.fn();
+
+const setup = (isLoading) => {
+  jest.clearAllMocks();
+  React.useState = jest.fn(() => [isLoading, mockSetisLoading]);
+  return mount(<EditArmyPage match={{ params: { ID: '5555' } }} />);
+};
+
+const mockRoles = [{ id: '1111', Name: 'HQ', SortOrder: 0 }];
+const mockUser = { uid: '2222' };
+const mockCodex = 'zzzzzzz';
+const mockSquad = [{
+  id: 'TigerSquad', Codex: 'zzzzzzz', Name: 'PirateSquad11', Role: '1111', MinSize: 1, MaxSize: 5, Units: ['TigerLeader'],
+}];
+const mockUnit = [{ id: 'TigerLeader', Name: 'PirateUnit11', Gear: ['33333', '44444'] }];
+const mockEquipment = [{ id: '33333', Name: 'mockSword' }, { id: '44444', Name: 'mockGun' }];
```

Comment:
All of these mocks feel really cluttered here. Probably should be in the before each block.